queries/foam/highlights.scm

```diff
@@ -0,0 +1,61 @@
+;; Comments
+(comment) @comment
+
+;; Generic Key-value pairs and dictionary keywords
+(key_value
+    keyword: (identifier) @function
+)
+(dict
+    key: (identifier) @type
+)
+
+;; Macros
+(macro
+    "$" @conditional
+    (prev_scope)* @conditional
+    (identifier)* @namespace
+)
+
+
+;; Directives
+"#" @conditional
+(preproc_call
+    directive: (identifier)* @conditional
+    argument: (identifier)* @namespace
+)
+(
+    (preproc_call
+        argument: (identifier)* @namespace
+    ) @conditional
+    (#match? @conditional "ifeq")
+)
+(
+    (preproc_call) @conditional
+    (#match? @conditional "(else|endif)")
+)
+
+;; Literal numbers and strings
+(number_literal) @float
+(string_literal) @string
+(escape_sequence) @escape
+
+;; Treat [m^2 s^-2] the same as if it was put in numbers format
+(dimensions dimension: (identifier) @float)
+
+;; Punctuation
+[
+  "("
+  ")"
+  "["
+  "]"
+  "{"
+  "}"
+  "#{"
+  "#}"
+  ";"
+] @punctuation
+
+;; Special identifiers
+([(identifier) "on" "off" "true" "false" "yes" "no"] @attribute
+(#match? @attribute "^(uniform|non-uniform|and|or|on|off|true|false|yes|no)$")
```

Comment:
I don't think that all those are `@attribute`. Rather `@keyword.operator` `@constant.builtin` `@boolean`. Maybe `uniform`, `non-uniform`...

Reply:
I tried to stay generic here, as these queries will get used in a few different places; but if you think `@constant.builtin` is a better choice, I'll go for that; as it seems this capture will be widely supported too.